File: src/infrastructure/udp/services/host-identity-service.ts
import { injectable } from "@needle-di/core";
import { UDP_HOST_COUNTER_BASE, UDP_HOST_PEER_ID } from "../../../core/udp/constants/udp-host-identity-constants.ts";

/**
 * Identity the dedicated host presents on the p2p channel: the HOST's
 * character id (peers gate on it — §4 gate 1) and its counter base (feeds the
 * session key K = peer_base ^ host_base).
 */
@injectable()
export class HostIdentityService {
  readonly peerId = UDP_HOST_PEER_ID;
  readonly counterBase = UDP_HOST_COUNTER_BASE;
}

File: src/GameplayServer/Identity/HostIdentityService.cs
using Mgo2Server.Shared.Constants;

namespace Mgo2Server.GameplayServer.Identity;

/// <summary>
/// Identity the gameplay server presents on the peer-to-peer channel: its
/// character identifier, which the joining client gates on, and the counter
/// base that feeds the session key.
/// </summary>
public sealed class HostIdentityService
{
    /// <summary>Identifier of the character this host plays as.</summary>
    public uint PeerIdentifier => UdpHostIdentityConstants.HostPeerIdentifier;

    /// <summary>Counter base this host announces.</summary>
    public uint CounterBase => UdpHostIdentityConstants.HostCounterBase;
}
